db/seeds.rb

```diff
@@ -37,6 +37,11 @@ def audit_modified?(type, name, attributes = {})
 
 def create_filters(role, collection)
   collection.group_by(&:resource_type).each do |resource, permissions|
+    # check if the filter exists already
+    next if Filter.joins(:permissions).find_by("permissions.id" => permissions.map(&:id),
+                                               :role_id => role.id
```

Comment:
I don't think this is a good check, there might be filter for the same role with the same permission just with some condition or only applied to some organization/location.

After reading the original issue I think it might be better to fix [katello seeding](https://github.com/Katello/katello/blob/master/db/seeds.d/105-roles_permissions.rb#L64) so that it does not touch existing roles. Note that use could modify them. If later more permissions are added to the role and it's desirable to add it even on existing system, it should do done in [migrations](https://github.com/theforeman/foreman/pull/3592/files#diff-c22e283e1a76f6464c5abea3aacbd56cR1) using fake models.